src/Pact/Analyze/Types/Languages.hs

```diff
@@ -160,6 +167,23 @@ data Core t a where
   -- for NOT, and two operands for AND or OR.
   Logical :: LogicalOp -> [t Bool] -> Core t Bool
 
+instance (UserShow a, UserShow (t a), UserShow (t Integer), UserShow (t Decimal), UserShow (t Time))
+  => UserShow (Core t a) where
+  userShowsPrec d = \case
+    Lit a                     -> userShowsPrec d a
+    Sym s                     -> tShow s
+    StrConcat x y             -> parenList ["+", userShow x, userShow y]
+    StrLength _str            -> parenList ["length", "TODO"]
```

Comment:
changing this line to `parenList ["length", userShow str]` and adding this constraint `UserShow (t String)` gives:

```
/home/joel/code/github/kadena-io/pact/src/Pact/Analyze/
Types/Languages.hs:170:10: error:
    * The constraint `UserShow (t String)'
        is no smaller than the instance head
      (Use UndecidableInstances to permit this)
    * In the instance declaration for `UserShow (Core 
t a)'
```

I need to think of a better way to do this.